src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala

```diff
@@ -149,21 +161,61 @@ object macros:
     
         case '[JsonObject] =>
           Json.fromJsonObject(JsonObjectUnit)
+
+        case '[Json] =>
+          JsonUnit
+
+        case '[JsonNumber] =>
+          Json.fromJsonNumber(JsonNumberUnit)
     
         case '[t] if TypeRepr.of[t] <:< TypeRepr.of[Product] =>
```

Comment:
Let's organise the case in a following manner:
- unit
- simple int values
- simple real values
- char and string
- big ints
- big reals
- uuid
- json values
- products